redaxo/src/core/lib/util/i18n.php

```diff
@@ -81,16 +84,19 @@ public static function getLanguage()
     public static function addDirectory($dir)
     {
         $dir = rtrim($dir, DIRECTORY_SEPARATOR);
-
         if (in_array($dir, self::$directories, true)) {
             return;
         }
-
         self::$directories[] = $dir;
 
+        if (!is_readable($dir)) {
```

Comment:
wirft auch exceptions für Ordner, die nicht existieren.

Reply:
sollte man bei `rex_i18n::addDirectory($dir)` voraussetzen, dass ein Verzeichnis vorhanden ist?

Reply:
nicht sicher ob wir das gleiche meinen, aber hier https://github.com/redaxo/redaxo/pull/1925#discussion_r219703584 hatte ich mich auch für eine exception ausgesprochen.

Reply:
https://github.com/redaxo/redaxo/blob/ffc43e245de91d40873303204f38c618ed448f16/redaxo/src/core/lib/util/i18n.php#L92-L94

dort wird jetzt eine exception geworfen, egal ob der Ordner existiert oder nicht lesbar ist. Dies sorgt aktuell dafür, dass bei jedem Package, dass keinen lang Ordner hat eine Exception geworfen wird. (siehe)
https://github.com/redaxo/redaxo/blob/ffc43e245de91d40873303204f38c618ed448f16/redaxo/src/core/lib/packages/package.php#L313-L318

Wenn ich dich richtig verstehe, sollte man bei `package::enlist` prüfen, ob das Verzeichnis existiert? Oder sollte so etwas in rex_i18n::addDirectory($dir) passieren?

Reply:
Hm ggf, können wir dann den check nicht machen.

Unser ziel ist es ja is-readable/file-exists etc calls zu vermeiden

Reply:
Arggg, stimmt. Wäre es dann nicht sinniger das `is_readable` gegen `file_exists` zu tauschen, da man ja i.d.R. auch wenn Besitzer/Gruppe nicht stimmen, zumindest lesen darf. Oder hast du andere Ideen?